### PR TITLE
Restructure docs and add deletion info

### PR DESCRIPTION
**Pull request by tobru**
This PR restructures the docs "a bit", cleans up old and wrong stuff and adds some documentation about the deletion stuff.

@srueg and @Kidswiss Please check if I really got this right with the deletion and that the wording is correct. I would also appreciate another look at the docs: Do we cover all aspects of the Operator? Are any important concepts missing which need documentation? I feel we miss some things, but can't currently say what.

### CODE REVIEW

File: docs/modules/ROOT/pages/explanation/deletion.adoc [hunk 1]
```diff
@@ -0,0 +1,34 @@
+= Object Deletion
+
+Object deletion is considered a dangerous activity which could lead into dataloss, therefore Lieutenant Operator implements a safeguard and a configuration per object what to do with external resources.
+
+== Deletion Protection
+
+The annotation `syn.tools/protected-delete` controls if an object can be deleted or not. As long as this annotation holds the value `true`, the finalizer will block the object from being deleted.
+
+Objects generated by the Operator (mainly the `GitRepo` object) are annotated as configured in the environment variable `LIEUTENANT_DELETE_PROTECTION` (see xref:references/configuration.adoc[References/Configuration]).
```
Comment (srueg) [inline docs/modules/ROOT/pages/explanation/deletion.adoc:9]:
```suggestion
The Operator automatically annotates objects as configured in the environment variable `LIEUTENANT_DELETE_PROTECTION` (see xref:references/configuration.adoc[References/Configuration]).
```

Not just objects it generates, also objects that were created by another tool.

File: docs/modules/ROOT/pages/explanation/deletion.adoc [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+[cols=",,",options="header",]
+|===
+
+|Policy
+|Git repo
+|Vault secret
+
+|_Archive_
+|Archival of Git repository
+|Secret soft deletion
```
Comment (srueg) [inline docs/modules/ROOT/pages/explanation/deletion.adoc:24]:
```suggestion
|Secret https://www.vaultproject.io/docs/secrets/kv/kv-v2#deleting-and-destroying-data[soft deletion]
```

File: docs/modules/ROOT/pages/explanation/deletion.adoc [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+|_Delete_
+|Deletion of Git repository
+|Secret hard deletion
```
Comment (srueg) [inline docs/modules/ROOT/pages/explanation/deletion.adoc:28]:
```suggestion
|Secret https://www.vaultproject.io/docs/secrets/kv/kv-v2#deleting-and-destroying-data[hard deletion]
```

File: docs/modules/ROOT/pages/explanation/design.adoc [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+Lieutenant manages the CR objects and queries the status fields to get
+the status.
+
+The Operator manages the following objects:
+
+_GitRepo_
+
+* Create Git repository +
+** By default on http://git.vshn.net/[git.vshn.net] GitLab
+** Supported are GitLab, GitHub and Gitea APIs
```
Comment (srueg) [inline docs/modules/ROOT/pages/explanation/design.adoc:46]: Should we mention that only GitLab is implemented to far?

File: docs/modules/ROOT/pages/how-tos/create-cluster.adoc [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+metadata:
+  name: c-ae3os1
+  namespace: lieutenant
+  annotations:
+    syn.tools/protected-delete: "false"
+spec:
+  displayName: Another Big Corp. Production Cluster
+  deletionPolicy: Delete
+  gitRepoTemplate:
+    path: cluster
```
Comment (srueg) [inline docs/modules/ROOT/pages/how-tos/create-cluster.adoc:18]:
```suggestion
    path: cluster-catalogs # path (org/group) to repo
```

File: docs/modules/ROOT/pages/how-tos/create-cluster.adoc [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    repoName: cluster2
```
Comment (srueg) [inline docs/modules/ROOT/pages/how-tos/create-cluster.adoc:19]:
```suggestion
    repoName: cluster2 # name of the repo to create
```

File: docs/modules/ROOT/pages/how-tos/create-cluster.adoc [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+    deletionPolicy: Delete
+    apiSecretRef:
```
Comment (srueg) [inline docs/modules/ROOT/pages/how-tos/create-cluster.adoc:21]:
```suggestion
    apiSecretRef: # reference to a secret containing credentials for the git provider
```

File: docs/modules/ROOT/pages/how-tos/create-gitrepo.adoc [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+....
+apiVersion: syn.tools/v1alpha1
+kind: GitRepo
+metadata:
+  name: example-gitrepo2
+  namespace: lieutenant
+spec:
+  tenantRef:
+    name: foo
+    namespace: lieutenant
```
Comment (srueg) [inline docs/modules/ROOT/pages/how-tos/create-gitrepo.adoc:17]:
```suggestion
```

Needs to be in the same namespace.

File: docs/modules/ROOT/pages/how-tos/create-gitrepo.adoc [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+  apiSecretRef:
+    name: lieutenant-secret
+  path: cluster/subgroup
+  repoName: bar
+  deployKeys:
+    test:
+      type: ssh-ed25519
+      key: AAAACxxxx
+      writeAccess: true
+      writeAccess: false
```
Comment (srueg) [inline docs/modules/ROOT/pages/how-tos/create-gitrepo.adoc:27]:
```suggestion
```

File: docs/modules/ROOT/pages/how-tos/create-tenant.adoc [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+spec:
+  displayName: Big Corp.
+  gitRepoTemplate:
+    path: tenant
+    repoName: tenant1
+    deletionPolicy: Delete
+    apiSecretRef:
+      name: lieutenant-secret
+      namespace: lieutenant
+    deployKeys:
```
Comment (srueg) [inline docs/modules/ROOT/pages/how-tos/create-tenant.adoc:21]: No deploy keys are needed on tenant repos so far.

File: docs/modules/ROOT/pages/how-tos/gitlab-connection.adoc [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+  endpoint: http://10.144.1.197:8080
+  token: Bvi1s19qp25Wt-rtGfMy
+kind: Secret
+metadata:
+  name: lieutenant-secret
+  namespace: syn-lieutenant
+type: Opaque
+....
+
+Save the file as `secret.yaml` and then apply it with `kubectl apply -f secret.yaml`.
```
Comment (srueg) [inline docs/modules/ROOT/pages/how-tos/gitlab-connection.adoc:30]: I'd prefer a copy pastable command:
```
kubectl -n lieutenant create secret generic lieutenant-secret \
  --from-literal endpoint=http://10.144.1.197:8080 \
  --from-literal token=<token>
```

File: docs/modules/ROOT/pages/how-tos/gitlab-connection.adoc [hunk 12]
```diff
@@ -0,0 +1,30 @@
+= Connection to GitLab
+
+For the Lieutenant Operator to be able to connect to the GitLab API, the following configuration needs to be made.
+
+== Get GitLab Token
+
+Visit the GitLab instance you'd like to use. Login with the user that has the permissions necessary to write to the group you want to store your Project Syn repositories.
```
Comment (srueg) [inline docs/modules/ROOT/pages/how-tos/gitlab-connection.adoc:7]: Maybe a numbered list 1., 2. 3. would make more sense here?

File: docs/modules/ROOT/pages/how-tos/installation.adoc [hunk 13]
```diff
@@ -0,0 +1,16 @@
+= Installation of the Operator
+
+The Operator provides a Kustomize folder which allows for easy installation:
+
+[source,shell]
+--
+kubectl create namespace lieutenant
+
+# CRDs (global scope)
+kubectl apply -k github.com/projectsyn/lieutenant-operator/deploy/crds
```
Comment (srueg) [inline docs/modules/ROOT/pages/how-tos/installation.adoc:10]: Is there a way in Antora to define a version number in a variable?
Then we could use something like

```suggestion
kubectl apply -k github.com/projectsyn/lieutenant-operator/deploy/crds?ref=${antora_version}
```

Comment (tobru) [reply docs/modules/ROOT/pages/how-tos/installation.adoc:10]: That's a good idea, but at the moment we're not yet versioning the documentation and are always pointing to the master branch. This is something we certainly need to do in the future.

**Pull request merged**